### PR TITLE
feat: workflow js run on browser

### PR DESCRIPTION
**Pull request by ayushpahwa**
## Description
This PR introduces browser run for workflows JS files. 
Changes:
- In case of workflows js object, there are no variables in the definition. This causes the jsaction.variables to be undefined which causes further issues down the line preventing second API call for update to the server. Added handling for that.
- Pushed workflows action to the data tree to start with linting and autocomplete. This was coming as empty for workflows editor.

#### PR fixes following issue(s)
Fixes # (issue number)

#### Type of change

- New feature (non-breaking change which adds functionality)

- [ ] Manual
- [ ] JUnit
- [ ] Jest
- [ ] Cypress
>
>
#### Dev activity
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

#### QA activity:
- [ ] [Speedbreak features](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#speedbreakers-) have been covered
- [ ] Test plan covers all impacted features and [areas of interest](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#areas-of-interest-)
- [ ] Test plan has been peer reviewed by project stakeholders and other QA members
- [ ] Manually tested functionality on DP
- [ ] We had an implementation alignment call with stakeholders post QA Round 2
- [ ] Cypress test cases have been added and approved by SDET/manual QA
- [ ] Added `Test Plan Approved` label after Cypress tests were reviewed
- [ ] Added `Test Plan Approved` label after JUnit tests were reviewed

## Summary by CodeRabbit

- **New Features**
	- Enhanced data selection capabilities to include workflow actions and JavaScript actions.
- **Bug Fixes**
	- Improved stability in JavaScript action handling by ensuring variables are always defined, preventing potential errors.

### CODE REVIEW

File: app/client/src/utils/JSPaneUtils.tsx [hunk 1]
```diff
@@ -134,8 +134,9 @@ export const getDifferenceInJSCollection = (
       jsAction.actions.splice(deleteArchived, 1);
     }
   }
-  //change in variables
-  const varList = jsAction.variables;
+  //change in variables. In cases the variable list is not present, jsAction.variables will be undefined
+  // we are setting to empty array to avoid undefined errors further in the code (especially in case of workflows main file)
+  const varList = jsAction.variables || [];
```
Comment (ApekshaBhosale) [inline app/client/src/utils/JSPaneUtils.tsx:139]: @ayushpahwa what's the value backend sending to client for variables prop?

**Review by ApekshaBhosale — CHANGES_REQUESTED** [review]

File: app/client/src/utils/JSPaneUtils.tsx (hunk 1 above)
Comment (ayushpahwa) [reply app/client/src/utils/JSPaneUtils.tsx:139]: Nope, it is not coming from server. 
image

Comment (ayushpahwa) [reply app/client/src/utils/JSPaneUtils.tsx:139]: image
This is how it looks for a normal jsObject

**Comment by marks0351** [comment]
Removing myseleves as a reviewer, let me know if there are any specific inputs needed from me.

**Review by trishaanand — APPROVED** [review]
Approving on behalf of Rishabh who approved only the EE PR which is a superset of all changes in this PR.

**Pull request merged**